src/renderer/components/input/input.tsx

```diff
@@ -153,7 +161,7 @@ export class Input extends React.Component<InputProps, State> {
   async validate() {
     const value = this.getValue();
     let validationId = (this.validationId = ""); // reset every time for async validators
-    const asyncValidators: Promise<any>[] = [];
+    const asyncValidators: Promise<React.ReactNode>[] = [];
```

Comment:
Could you please explain why is validators an array of `Promise<React.ReactNode>`? I don't go through the call but don't get why they are `Promise<React.ReactNode>`

Reply:
The reason is on line 182 below. Basically this is here to signify that the promises added to this list should resolve to something that can be rendered (the error message)